CMakeLists.txt

```diff
@@ -4,6 +4,13 @@ project( googletest-distribution )
 
 enable_testing()
 
+include(GNUInstallDirs)
+
+option(SKIP_GOOGLETEST_INSTALL "Skip installation" ${SKIP_GOOGLETEST_INSTALL} )
+if( NOT SKIP_GOOGLETEST_INSTALL )
+   set( ENABLE_GOOGLETEST_INSTALL ON )
```

Comment:
No, `option` doesn't work that way. To prove it, try this trivial test:

```
$ mkdir /tmp/cmake-option && cd /tmp/cmake-option
$ echo 'option(FOO "This is option FOO" OFF)' > CMakeLists.txt
$ cmake -DFOO=ON .
$ grep FOO CMakeCache.txt
```

...and you'll see that `FOO` is `ON`. `option` is not `set`; if the option already has a value, the *initial* value does not override it.

(Caveat: `option` will only look for an existing *cache* value. A project embedding Google Test that doesn't want to install it needs to say `set(SKIP_GOOGLETEST_INSTALL ON CACHE "" INTERNAL)`; `set(SKIP_GOOGLETEST_INSTALL ON)` won't work. I'm not sure that's worth employing this anti-pattern, however.)